docs/rules/no-deprecated-functions.md

```diff
@@ -0,0 +1,43 @@
+# Warns on usage of deprecated functions (no-deprecated-functions)
+
+Over the years jest has accrued some debt in the form of functions that have
+either been renamed for clarity, or replaced with more powerful APIs.
+
+While typically these deprecated functions are kept in the codebase for a number
+of majors, eventually they are removed completely.
+
+## Rule details
+
+This rule warns about calls to deprecated functions, and provides details on
+what to replace them with.
+
+This rule can also autofix a number of these deprecations for you.
+
+### `require.requireActual.` & `require.requireMock`
+
+These functions were removed in Jest 26.
+
+Originally in the early days of jest, the `requireActual`& `requireMock`
+functions were placed onto the `require` function.
+
+These functions were later moved onto the `jest` global, and their use via
+`require` deprecated. Finally, the release of Jest 26 saw them removed from the
+`require` function all together.
+
+The PR implementing the removal can be found
+[here](https://github.com/facebook/jest/pull/9854).
+
+### `jest.addMatchers`
+
+This function has been replaced with `expect.extend`, and will ideally be
```

Comment:
main issue with `addMatchers` is that it's just a proxy to `jasmine.addMatchers`, and we want people to use `expect` and `expect.extend`, like you mention, directly.

It will throw an error if using `jest-circus`

Reply:
Might as well keep things simple if we're replacing `jasmine` with `jest-circus` 🤷 

I assume that you can call `jasmine.addMatchers` directly if you were really wanted to?

Reply:
yup! (unless you're using circus)